lab3.py

```diff
@@ -0,0 +1,31 @@
+import subprocess
+import sys
+import boto3
+import datetime
+
+s3 = boto3.client('s3')
+
+
+def main(argv):
+    cmd = argv
+    log_file_name = datetime.datetime.now().strftime("%m_%d_%Y") + "_logfile"
+    kickoff_subprocess(cmd, log_file_name)
+    upload_output_to_s3(log_file_name)
+
+
+def kickoff_subprocess(cmd, log_file_name):
+    process = subprocess.call(cmd, shell=True)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It appears that you are using a string to construct a shell command that invokes the shell explicitly, via `shell=True`. This might be susceptible to shell injection. We recommend that you use a `shlex.quote()` function to properly escape whitespace and shell meta-characters in strings that are used to construct shell commands.

[Learn more](https://docs.python.org/3/library/subprocess.html#security-considerations)